relay-server/src/services/project_cache.rs

```diff
@@ -913,6 +895,48 @@ impl ProjectCacheBroker {
         }
     }
 
+    /// Returns backoff timeout for an unspool attempt.
+    fn next_unspool_attempt(&mut self) -> Duration {
+        self.config.spool_envelopes_unspool_interval() + self.buffer_unspool_backoff.next_backoff()
+    }
+
+    fn schedule_unspool(&mut self) {
+        if self.buffer_unspool_handle.is_idle() {
+            // Set the time for the next attempt.
+            let wait = self.next_unspool_attempt();
+            self.buffer_unspool_handle.set(wait);
+        }
+    }
+
+    /// Iterates the buffer index and tries to unspool the envelopes if the project state is still
+    /// valid.
+    ///
+    /// This makes sure we always moving the unspool forward, even if we do not fetch the project
+    /// states updates, but still can process data based on the existing cache.
+    fn handle_periodic_unspool(&mut self) {
+        self.buffer_unspool_handle.reset();
+        let keys = self.index.keys().cloned().collect::<Box<[_]>>();
+
+        // If there is nothing spooled, schedule the next check a little bit later.
+        if keys.is_empty() {
+            self.schedule_unspool();
+        }
+
+        for project_key in keys.iter() {
+            if self
+                .projects
+                .get(project_key)
+                .map_or(false, |state| state.valid_state().is_some())
```

Comment:
It will be refreshed as it was before like normal refresh operation. 
Here I just want to check for the valid states, and if there are try to unspool them.

Reply:
Gotcha. It might not be related to the goal of this PR but I feel like we should trigger a prefetch at this point. Otherwise we still depend on incoming traffic to get old data unspooled, right?

Reply:
it's a valid point. 
I've switched to using `get_cached_state` instead, which will trigger the refresh of the project state if it's not valid or expired.